derive_client/_bridge/client.py

```diff
@@ -31,181 +25,394 @@
+            params[1],  # receiver
+            params[2],  # amountLD
+            params[3],  # minAmountLD
+            params[4],  # extraOptions
+            b"",  # composeMsg
+            b"",  # oftCmd
+        )
+        fees = token_contract.functions.quoteSend(
+            send_params,  # params, feeParams
+            False,  # payInLzToken
+        ).call()
 
-        address = self.config.contracts.L1_CHUG_SPLASH_PROXY
-        bridge_abi = json.loads(L1_STANDARD_BRIDGE_ABI_PATH.read_text())
-        proxy_contract = get_contract(w3=w3, address=address, abi=bridge_abi)
+        native_fee, lz_token_fee = fees
+        _refundAddress = self.owner
 
-        tx = prepare_mainnet_to_derive_gas_tx(w3=w3, account=self.account, amount=amount, proxy_contract=proxy_contract)
-        tx_result = send_and_confirm_tx(w3=w3, tx=tx, private_key=self.account._private_key, action="bridgeETH()")
+        # fees = token_contract.functions.quoteOFT(
+        #     send_params,  # params, feeParams
+        #     # False,  # payInLzToken
+        # ).call()
+        # breakpoint()
+
+        func = token_contract.functions.send(send_params, fees, _refundAddress)  # lzTokenFee
+
+        tx = build_standard_transaction(func=func, account=self.account, w3=self.w3, value=native_fee)
+
+        tx_result = send_and_confirm_tx(w3=self.w3, tx=tx, private_key=self.private_key, action="bridge()")
         return tx_result
 
-    def withdraw_with_wrapper(
+    def deposit(
         self,
         amount: int,
-        receiver: Address,
-        token_data: MintableTokenData,
-        wallet: Address,
-        private_key: str,
+        token_data: NonMintableTokenData | MintableTokenData,
     ) -> TxResult:
         """
-        Checks if sufficent gas is available in derive, if not funds the wallet.
-        Prepares, signs, and sends a withdrawal transaction using the withdraw wrapper.
+        Deposit funds by preparing, signing, and sending a bridging transaction.
         """
 
+        spender = token_data.Vault if token_data.isNewBridge else self.deposit_helper.address
+        token_contract = get_erc20_contract(self.w3, token_data.NonMintableToken)
+        ensure_balance(token_contract, self.owner, amount)
+        ensure_allowance(
+            w3=self.w3,
+            token_contract=token_contract,
+            owner=self.owner,
+            spender=spender,
+            amount=amount,
+            private_key=self.private_key,
+        )
+
+        if token_data.isNewBridge:
+            tx = self._prepare_new_style_deposit(token_data, amount, self.wallet)
+        else:
+            tx = self._prepare_old_style_deposit(token_data, amount)
+
+        tx_result = send_and_confirm_tx(w3=self.w3, tx=tx, private_key=self.private_key, action="bridge()")
+        return tx_result
+
+    def withdraw_drv(
+        self,
+        amount: int,
+        target_chain: int,
+    ):
+        self._ensure_derive_eth_balance()
```

Comment:
This was added to ensure gas deposit in case trying to withdraw DRV (rather than super socket methods, for which this was already being called). fcf88ac67f9d250cb55a2c24715c352a73ef1a7c